src/getTxnSig.c

```diff
@@ -0,0 +1,314 @@
+            }
+            else {
+                os_memmove(ctx->partialAmountStr, ctx->amountStr + ctx->displayIndex, ctx->amountLength);
+            }
+            UX_REDISPLAY();
+            break;
+
+        case BUTTON_EVT_RELEASED | BUTTON_LEFT | BUTTON_RIGHT: // PROCEED
+            bin2dec(ctx->fromShardStr, ctx->txContent.fromShard);
+            bin2dec(ctx->toShardStr, ctx->txContent.toShard);
+            UX_DISPLAY(ui_fromshard_approve, NULL);
+            break;
+    }
+    // (The return value of a button handler is irrelevant; it is never
+    // checked.)
+    return 0;
+}
+
+static const bagl_element_t ui_amount_compare[] = {
+        UI_BACKGROUND(),
+
+        UI_ICON_LEFT(0x00, BAGL_GLYPH_ICON_CROSS),
+        UI_ICON_RIGHT(0x00, BAGL_GLYPH_ICON_CHECK),
+
+        UI_TEXT(0x00, 0, 12, 128, "Amount:"),
+        UI_TEXT(0x00, 0, 26, 128, global.signTxnContext.partialAmountStr),
+};
+
+static unsigned int ui_amount_compare_button(unsigned int button_mask, unsigned int button_mask_counter) {
+    switch (button_mask) {
+        case BUTTON_EVT_RELEASED | BUTTON_LEFT: // REJECT
+            io_exchange_with_code(SW_USER_REJECTED, 0);
+            // Return to the main screen.
+            ui_idle();
+            break;
+
+        case BUTTON_EVT_RELEASED | BUTTON_RIGHT: // APPROVE
+            bin2dec(ctx->fromShardStr, ctx->txContent.fromShard);
+            bin2dec(ctx->toShardStr, ctx->txContent.toShard);
+            UX_DISPLAY(ui_fromshard_approve, NULL);
+            break;
+    }
+    return 0;
+}
+
+static const bagl_element_t ui_address_compare[] = {
+        UI_BACKGROUND(),
+
+        UI_ICON_LEFT(0x01, BAGL_GLYPH_ICON_LEFT),
+        UI_ICON_RIGHT(0x02, BAGL_GLYPH_ICON_RIGHT),
+
+        UI_TEXT(0x00, 0, 12, 128, "Send to Addr:"),
+        UI_TEXT(0x00, 0, 26, 128, global.signTxnContext.partialAddrStr),
+};
+
+
+static const bagl_element_t* ui_prepro_address_compare(const bagl_element_t *element) {
+    switch (element->component.userid) {
+        case 1:
+            return (ctx->displayIndex == 0) ? NULL : element;
+        case 2:
+            return (ctx->displayIndex == sizeof(ctx->toAddr)-12) ? NULL : element;
+        default:
+            return element;
+    }
+}
+
+static unsigned int ui_address_compare_button(unsigned int button_mask, unsigned int button_mask_counter) {
+    uint8_t numberBuf[32];
+
+    switch (button_mask) {
+        case BUTTON_LEFT:
+        case BUTTON_EVT_FAST | BUTTON_LEFT: // SEEK LEFT
+            // Decrement the displayIndex when the left button is pressed (or held).
+            if (ctx->displayIndex > 0) {
+                ctx->displayIndex--;
+            }
+
+            os_memmove(ctx->partialAddrStr, ctx->toAddr+ctx->displayIndex, 12);
+            // Re-render the screen.
+            UX_REDISPLAY();
+            break;
+
+        case BUTTON_RIGHT:
+        case BUTTON_EVT_FAST | BUTTON_RIGHT: // SEEK RIGHT
```

Comment:
Why `BUTTON_LEFT`, `BUTTON_RIGHT` not directly folded into their following case, like you did with `  case BUTTON_EVT_RELEASED | BUTTON_LEFT | BUTTON_RIGHT:` Is it for future code explicit for button_left, button_right?